apis/cloudfront/v1alpha1/zz_cloud_front_origin_access_i_dentity.go

```diff
@@ -0,0 +1,94 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// CloudFrontOriginAccessIDentityParameters defines the desired state of CloudFrontOriginAccessIDentity
+type CloudFrontOriginAccessIDentityParameters struct {
+	// Region is which region the CloudFrontOriginAccessIDentity will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+	// The current configuration information for the identity.
+	// +kubebuilder:validation:Required
+	CloudFrontOriginAccessIDentityConfig           *OriginAccessIDentityConfig `json:"cloudFrontOriginAccessIDentityConfig"`
+	CustomCloudFrontOriginAccessIDentityParameters `json:",inline"`
+}
+
+// CloudFrontOriginAccessIDentitySpec defines the desired state of CloudFrontOriginAccessIDentity
+type CloudFrontOriginAccessIDentitySpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+	ForProvider       CloudFrontOriginAccessIDentityParameters `json:"forProvider"`
+}
+
+// CloudFrontOriginAccessIDentityObservation defines the observed state of CloudFrontOriginAccessIDentity
+type CloudFrontOriginAccessIDentityObservation struct {
+	// The origin access identity's information.
+	CloudFrontOriginAccessIDentity *OriginAccessIDentity `json:"cloudFrontOriginAccessIDentity,omitempty"`
+	// The current version of the origin access identity created.
+	ETag *string `json:"eTag,omitempty"`
+	// The fully qualified URI of the new origin access identity just created.
+	Location *string `json:"location,omitempty"`
+}
+
+// CloudFrontOriginAccessIDentityStatus defines the observed state of CloudFrontOriginAccessIDentity.
+type CloudFrontOriginAccessIDentityStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+	AtProvider          CloudFrontOriginAccessIDentityObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// CloudFrontOriginAccessIDentity is the Schema for the CloudFrontOriginAccessIDentities API
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="EXTERNAL-NAME",type="string",JSONPath=".metadata.annotations.crossplane\\.io/external-name"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,aws}
+type CloudFrontOriginAccessIDentity struct {
```

Comment:
Can we avoid this mangled type name? Presuming doing so would require upstream ACK changes, perhaps we could temporarily disable generation of this file, manually `s/IDentity/Identity/g` across this implementation, and add a note somewhere tracking that generation is disabled?

Reply:
This is fixed in ack-generate via https://github.com/aws-controllers-k8s/code-generator/pull/107, but we are using an older version until https://github.com/crossplane/provider-aws/pull/920 is merged. 

I've started the process of ignoring originaccessidentity and checking in the files manually, but there are some conflicts with other CloudFront API definitions. Still working on this to see if we can make it work with correct capitalization.

Reply:
After working on the code a bit and talking to @AaronME, I'd like to wait to get #920 merged instead of hacking around code-generation, as several CloudFront Managed Resources would be modified by the change.

Reply:
@stevendborrelli #920 is merged ;)

Reply:
@haarchri Thanks! I will update the PR.